### PR TITLE
New neuroscience packages: py-bmtk, py-neurotools

### PR DESCRIPTION
**Pull request by heerener**

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed the 3 sha256.

File: var/spack/repos/builtin/packages/py-bmtk/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    depends_on("py-jsonschema", type=("run"))
+    depends_on("py-pandas", type=("run"))
+    depends_on("py-numpy", type=("run"))
+    depends_on("py-six", type=("run"))
+    depends_on("py-h5py", type=("run"))
+    depends_on("py-matplotlib", type=("run"))
+    depends_on("py-scipy", type=("run"))
+    depends_on("py-scikit-image", type=("run"))
+    depends_on("py-sympy", type=("run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bmtk/package.py]: These should be `type=("build", "run")` so the import tests pass

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    depends_on("py-numpy", type="run")
+    depends_on("py-matplotlib", type="run")
+    depends_on("py-urllib3", type="run")
+    depends_on("py-mpi4py", type="run")
+    depends_on("py-tables", type="run")
+    depends_on("py-pyaml", type="run")
+    # py-interval is py2 only and most probably not used
+    # depends_on('py-interval', type='run')
+
+    patch("neurotools-0.3.1.patch", when="@0.3.1")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py]: Can you provide a link to where this patch comes from or some discussion of why it is needed?

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+class PyNeurotools(PythonPackage):
+    """A collection of tools for representing and anlyzing neuroscientific
+    data."""
+
+    homepage = "http://neuralensemble.org/NeuroTools"
+    pypi = "neurotools/NeuroTools-0.3.1.tar.gz"
+
+    version("0.3.1", sha256="a459420fc0e9ff6b59af28716ddb0c75d11a63b8db80a5f4844e0d7a90c2c653")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py:17]: Missing build dep on setuptools

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    depends_on("py-scipy", type="run")
+    depends_on("py-numpy", type="run")
+    depends_on("py-matplotlib", type="run")
+    depends_on("py-urllib3", type="run")
+    depends_on("py-mpi4py", type="run")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py]: I don't see urllib3 or mpi4py imported anywhere

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    depends_on("py-tables", type="run")
+    depends_on("py-pyaml", type="run")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py]: All of these should be build/run

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    # py-interval is py2 only and most probably not used
+    # depends_on('py-interval', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py]: Spack no longer supports Python 2 so you can just remove this

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+class PyNeurotools(PythonPackage):
+    """A collection of tools for representing and anlyzing neuroscientific
+    data."""
+
+    homepage = "http://neuralensemble.org/NeuroTools"
+    pypi = "neurotools/NeuroTools-0.3.1.tar.gz"
+
+    version("0.3.1", sha256="a459420fc0e9ff6b59af28716ddb0c75d11a63b8db80a5f4844e0d7a90c2c653")
+
+    depends_on("py-setuptools", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py]:
```suggestion
    depends_on("py-setuptools", type="build")
```

File: var/spack/repos/builtin/packages/py-neurotools/package.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+    depends_on("py-tables", type=("build", "run"))
+    depends_on("py-pyaml", type=("build", "run"))
+
+    patch("neurotools-0.3.1.patch", when="@0.3.1")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-neurotools/package.py:25]: Still need a link to where this patch comes from or some discussion of why it is needed

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-neurotools/package.py:25]: Comes originally from https://github.com/BlueBrain/bbp-nixpkgs/pull/1026 (sorry, absolutely useless PR description)

The original code in essence tries to do this:
```
>>> np.random.RandomState().exponential(1.5, np.ceil(3.6))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 540, in numpy.random.mtrand.RandomState.exponential
  File "_common.pyx", line 630, in numpy.random._common.cont
TypeError: expected a sequence of integers or a single integer, got '4.0'
```
so patching in `int()` will avoid this error (the parameter in question are the size of the returned `numpy.array` containing random numbers).

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-neurotools/package.py:25]: Thanks, seems like the library is no longer under development so there's no need to report this to the developers

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**